### PR TITLE
Adds Interdyne Modsuits

### PR DESCRIPTION
**Pull request by YesterdaysPromise**
## About The Pull Request
Adds separate, Interdyne-brand Modsuits for Intrudyne pirates.  Incredibly fast and more techy, including among other things an organ thrower module (which is, based on flavour text, Interdyne tech). To balance this, they discharge dramatically fast, encouraging fast get-in get-out approach.

Sprite itself is mostly a combination of medical and syndicate suits, with labcoat bits for extra stylishness and evil feel. The colour pallete was enterily borrowed from Interdyne Pharmaceutics container sprite and Interdyne-produced E-surgery tools.
flayed
image
image

Lorewise, I borrowed a random idea from cyberpunk lore video I once watched and is now blurred for me now, upon seeing retrieval suits. Originally made by conjuction of Cybersun and Intrudyne, combining mechanics with supersoldier treatment for speed, allegadely for rapid response and retrieval off corpses. [By the way, it would be funny if someone coded an event where after cap's death on Revs or Cult, a bunch of guys in these suits would show up with sole goal of recovering the body and escaping with it (plus maybe some side syndicate objective)] Aaanyway, obviously, the rapid speed plays into a lot of Intrudyne-related tactics for the shady side of things as well. Much like a scalpel, their attacks are supposed to be fast, clean and precise.

## Why It's Good For The Game

Consistency thing, mostly. It doesn't make much sense that a medical corp would use their competitors tools when they could develop a counter. Also ties with organ thrower module, why would they have made it, if they didn't have modsuits for it?

## Changelog

:cl:
add: Added Interdyne ModSuits for Interdyne pirates.
/:cl:

### CODE REVIEW

**Comment by Pumpkinoe** [comment]
needs pictures!!

File: code/modules/mod/mod_theme.dm [hunk 1]
```diff
@@ -1159,6 +1159,88 @@
 	acid = 100
 	wound = 25
 
+/datum/mod_theme/interdyne
+	name = "interdyne"
+	desc = "A corpse-snatching and rapid-retrieval modsuit, resulting from a lucrative tech exchange between Intrudyne Pharmaceutics and Cybersun Industries."
```
Comment (carlarctg) [inline code/modules/mod/mod_theme.dm:1164]:
```suggestion
	desc = "A corpse-snatching and rapid-retrieval modsuit, resulting from a lucrative tech exchange between Interdyne Pharmaceutics and Cybersun Industries."
```

File: code/modules/mod/mod_theme.dm [hunk 2]
```diff
@@ -1159,6 +1159,88 @@
...
+	extended_desc = "While Woffle Co. and Azik Interstellar provide the means, Donk Co., Tiger Cooperative, Animal Rights Consortium and \
```
Comment (carlarctg) [inline code/modules/mod/mod_theme.dm:1165]:
```suggestion
	extended_desc = "While Waffle Co. and Azik Interstellar provide the means, Donk Co., Tiger Cooperative, Animal Rights Consortium and \
```

Bro whats with yo names

File: code/modules/mod/mod_theme.dm [hunk 3]
```diff
@@ -1159,6 +1159,88 @@
...
+		Gorlex Marauders willing or easly bribable brawn, S.E.L.F. and MI13 information, the clear syndicate tech providers would be Interdyne and Cybersun, \
+		their combined knowledge in technologies rivaled by only most enigmatic of aliens, and certainly not by any Nanostrasen scientist. \
+		This modsuit model is one of rare fruits created by their joint operations, mashing scrapped designs with super soldier enhancements. \
+		Already light, when powered on, this modsuits injects the wearer seemlessly with muscle-enhancing supplements, while adding piston strenght \
+		to their legs. Combination of these mechanisms is very energy draining - but results in next to no speed reduction for the wearer.\
+		Over the years, many a rich person, including Nanostrasen officials with premium subscriptions, had their life or genes rescued thanks to \
+		unrivaled speed of this suit model. Equally as many, however, mysteriously dissapeared in the flash of these white suits after they forgot \
+		to pay off said subscryptions in due time or posted unfavourable opinions on Intrudyne's gene-modding tech and ethics. "
```
Comment (carlarctg) [inline code/modules/mod/mod_theme.dm:1173]:
```suggestion
		to pay off said subscriptions in due time or publicly communicated unfavourable opinions on Interdyne's gene-modding tech and ethics. "
```

File: code/modules/mod/mod_theme.dm [hunk 4]
```diff
@@ -1159,6 +1159,88 @@
...
+			),
+			BOOTS_FLAGS = list(
+				UNSEALED_CLOTHING = THICKMATERIAL,
+				SEALED_CLOTHING = STOPSPRESSUREDAMAGE,
+				CAN_OVERSLOT = TRUE,
+			),
+		),
+	)
+
+/datum/armor/mod_theme_interdyne
```
Comment (carlarctg) [inline code/modules/mod/mod_theme.dm:1233]: That's pretty damn weak for an antagonist piece of armor. To contrast, the pirate coats the Jolly Roger gets are some of the best armor in the game, and they're spaceproof by default. Maybe give them the armor module syndies get?

File: code/modules/mod/mod_types.dm [hunk 5]
```diff
@@ -287,6 +287,18 @@
 		/obj/item/mod/module/visor/diaghud,
 	)
 
+
+/obj/item/mod/control/pre_equipped/interdyne
+	theme = /datum/mod_theme/interdyne
+	applied_cell = /obj/item/stock_parts/cell/super
+	applied_modules = list(
```
Comment (carlarctg) [inline code/modules/mod/mod_types.dm:294]: Where's the combat defib? Other modules might be interesting too. Med visor?

**Comment by Serxule** [comment]
how fast does the cell actually run out? also i think having the cell run out fast in exchange for no suit slowdown (something both syndi modsuits get via armor booster which doesnt drain the cell fast, maybe you could have it be a module that literally boosts the speed, making their movement faster than normal run speed in exchange for fast cell draining, like the gygax for example

File: code/modules/mod/mod_theme.dm (hunk 4 above)
Comment (YesterdaysPromise) [reply code/modules/mod/mod_theme.dm:1233]: buffed the speed and armour now.

File: code/modules/mod/mod_theme.dm (hunk 1 above)
Comment (YesterdaysPromise) [reply code/modules/mod/mod_theme.dm]: thank you for all your grammar checks - misspelling Interdyne as Intrudyne is one I corrected myself on quite often. now they all should be fixed.

**Comment by Sealed101** [comment]
curious choice of thigh armor plating

**Comment by SethLafuente** [comment]
Bababased

**Comment by SethLafuente** [comment]
If it's any useful I remember once looking through interdyne design pictures and someone send me this 

image

**Review by Wallemations — APPROVED** [review]

File: code/modules/mod/mod_theme.dm [hunk 6]
```diff
@@ -1159,6 +1159,88 @@
...
+		unrivaled speed of this suit model. Equally as many, however, mysteriously dissapeared in the flash of these white suits after they forgot \
+		to pay off said subscriptions in due time or publicly communicated unfavourable opinions on Intrudyne's gene-modding tech and ethics. "
+	default_skin = "interdyne"
+	armor_type = /datum/armor/mod_theme_interdyne
+	resistance_flags = FIRE_PROOF|ACID_PROOF
+	max_heat_protection_temperature = FIRE_SUIT_MAX_TEMP_PROTECT
+	charge_drain = DEFAULT_CHARGE_DRAIN * 2
+	slowdown_inactive = 0.0
+	slowdown_active = -0.5
+	inbuilt_modules = list(/obj/item/mod/module/quick_carry/advanced, /obj/item/mod/module/organ_thrower, /obj/item/mod/module/defibrillator/combat, /obj/item/mod/module/tether)
```
Comment (Fikou) [inline code/modules/mod/mod_theme.dm:1181]: why are these inbuilt if they are removable?

File: code/modules/mod/mod_theme.dm [hunk 7]
```diff
@@ -1159,6 +1159,88 @@
 	acid = 100
 	wound = 25
 
+/datum/mod_theme/interdyne
+	name = "interdyne"
+	desc = "A corpse-snatching and rapid-retrieval modsuit, resulting from a lucrative tech exchange between Interdyne Pharmaceutics and Cybersun Industries."
+	extended_desc = "While Waffle Co. and Azik Interstellar provide the means, Donk Co., Tiger Cooperative, Animal Rights Consortium and \
+		Gorlex Marauders willing or easly bribable brawn, S.E.L.F. and MI13 information, the clear syndicate tech providers would be Interdyne and Cybersun, \
+		their combined knowledge in technologies rivaled by only most enigmatic of aliens, and certainly not by any Nanostrasen scientist. \
+		This modsuit model is one of rare fruits created by their joint operations, mashing scrapped designs with super soldier enhancements. \
```
Comment (Fikou) [inline code/modules/mod/mod_theme.dm:1168]:
```suggestion
		their combined knowledge in technologies rivaled by only most enigmatic of aliens, and certainly not by any Nanotrasen scientist. \
		This model is one of rare fruits created by their joint operations, mashing scrapped designs with super soldier enhancements. \
```

File: code/modules/mod/mod_theme.dm [hunk 8]
```diff
@@ -1159,6 +1159,88 @@
...
+		to pay off said subscriptions in due time or publicly communicated unfavourable opinions on Intrudyne's gene-modding tech and ethics. "
+	default_skin = "interdyne"
+	armor_type = /datum/armor/mod_theme_interdyne
+	resistance_flags = FIRE_PROOF|ACID_PROOF
+	max_heat_protection_temperature = FIRE_SUIT_MAX_TEMP_PROTECT
+	charge_drain = DEFAULT_CHARGE_DRAIN * 2
+	slowdown_inactive = 0.0
+	slowdown_active = -0.5
+	inbuilt_modules = list(/obj/item/mod/module/quick_carry/advanced, /obj/item/mod/module/organ_thrower)
+	allowed_suit_storage = list(
```
Comment (san7890) [inline code/modules/mod/mod_theme.dm:1182]: can you alphabetize this list? there's extensions for vscode that will do it automatically for you with a simple keybind. just helps maintainability a lot

File: code/modules/mod/mod_types.dm [hunk 9]
```diff
@@ -287,6 +287,20 @@
...
```
Comment (san7890) [inline code/modules/mod/mod_types.dm:294]: ditto on alphabetize

**Review by san7890 — COMMENTED** [review]
two nits and i think we're solid?

**Comment by YesterdaysPromise** [comment]
> If it's any useful I remember once looking through interdyne design pictures and someone send me this
> 
> image

I was not aware this existed. Should I re-do the sprite in this style? People seem to not really like the cape anyway, so. Just want it confirmed that I should do it.

**Comment by Rex9001** [comment]
> > If it's any useful I remember once looking through interdyne design pictures and someone send me this
> 
> > 
> 
> > image
> 
> 
> 
> I was not aware this existed. Should I re-do the sprite in this style? People seem to not really like the cape anyway, so. Just want it confirmed that I should do it.

I think those might be sprites for a different server

**Review by Maurukas — APPROVED** [review]

**Pull request merged**